src/core/NgenSimulation.cpp

```diff
@@ -0,0 +1,254 @@
+#include <NgenSimulation.hpp>
+#include <NGenConfig.h>
+#include <Logger.hpp>
+
+#if NGEN_WITH_ROUTING
+#include "bmi/Bmi_Py_Adapter.hpp"
```

Comment:
```suggestion
#include "bmi/Bmi_Py_Adapter.hpp"
#include "utilities/parallel_utils.h"
```
Parallel utils is needed for t-route.